core/include/traccc/edm/track_state.hpp

```diff
@@ -81,26 +81,27 @@ struct track_state {
                       "The measurement dimension should be 1 or 2");
 
         matrix_type<D, 1> ret;
-        if (m_measurement.subs.get_indices()[0] == e_bound_loc0) {
-            matrix_operator().element(ret, 0, 0) = m_measurement.local[0];
-            if constexpr (D == 2u) {
-                matrix_operator().element(ret, 1, 0) = m_measurement.local[1];
-            }
-        } else if (m_measurement.subs.get_indices()[0] == e_bound_loc1) {
-            matrix_operator().element(ret, 0, 0) = m_measurement.local[1];
-            if constexpr (D == 2u) {
-                matrix_operator().element(ret, 1, 0) = m_measurement.local[0];
-            }
-        } else {
-            assert(
-                "The measurement index out of e_bound_loc0 and e_bound_loc1 "
-                "should not happen.");
-            matrix_operator().element(ret, 0, 0) = m_measurement.local[0];
-            if constexpr (D == 2u) {
-                matrix_operator().element(ret, 1, 0) = m_measurement.local[1];
-            }
+        switch (m_measurement.subs.get_indices()[0]) {
+            default:
```

Comment:
Why is this at the top and not at the bottom as is more common in switch statements?

Reply:
To be able to fall through it. I thought that was clear.

Indeed, this is not a standard setup. But it seemed like a neat way of achieving the current behaviour.

Whether we want to change the current behaviour is a discussion to be had though.